### PR TITLE
Signin signup signout

### PR DESCRIPTION
**Pull request by canvantuan12**

### CODE REVIEW

File: app/views/devise/registrations/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    <div class="field">
+      <%= f.label :email %><br />
+      <%= f.email_field :email, class: "form-control", required: true %>
+    </div>
+
+    <div class="field">
+      <%= f.label :password %>
+      <% if @minimum_password_length %>
+      <em>(<%= @minimum_password_length %> <%= t ".charactermini" %></em>
+      <% end %><br />
```
Comment (canvantuan12) [inline app/views/devise/registrations/new.html.erb]: indent

File: app/views/devise/sessions/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  <%= form_for(resource, as: resource_name,
+    url: session_path(resource_name)) do |f| %>
+    <div class="form-group">
+      <%= f.label :email %><br />
+      <%= f.email_field :email, autofocus: true,
+        class: "form-control", required: true %>
+    </div>
+
+    <div class="form-group">
+      <%= f.label :password %><br />
```
Comment (canvantuan12) [inline app/views/devise/sessions/new.html.erb]: indent

File: app/views/devise/shared/_links.html.erb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+<%- if controller_name != 'sessions' %>
+  <%= link_to "Log in", new_session_path(resource_name) %><br />
```
Comment (canvantuan12) [inline app/views/devise/shared/_links.html.erb]: indent

File: app/views/devise/shared/_links.html.erb [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+<% end -%>
+
+<%- if devise_mapping.registerable? && controller_name != 'registrations' %>
+  <%= link_to "Sign up", new_registration_path(resource_name) %><br />
```
Comment (canvantuan12) [inline app/views/devise/shared/_links.html.erb]: indent

File: app/views/devise/shared/_links.html.erb [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+<% end -%>
+
+<%- if devise_mapping.omniauthable? %>
+  <%- resource_class.omniauth_providers.each do |provider| %>
+    <%= link_to "Sign in with #{OmniAuth::Utils.camelize(provider)}",
+      omniauth_authorize_path(resource_name, provider) %><br />
```
Comment (canvantuan12) [inline app/views/devise/shared/_links.html.erb]: indent

File: config/environments/development.rb [hunk 6]
```diff
@@ -28,6 +28,7 @@
 
   # Don't care if the mailer can't send.
   config.action_mailer.raise_delivery_errors = false
+  config.action_mailer.default_url_options = { host: 'localhost', port: 3000 }
```
Comment (canvantuan12) [inline config/environments/development.rb]: indent

File: config/environments/development.rb [hunk 7]
```diff
@@ -28,6 +28,7 @@
 
   # Don't care if the mailer can't send.
   config.action_mailer.raise_delivery_errors = false
+  config.action_mailer.default_url_options = {host: 'localhost', port: 3000}
```
Comment (dieunb) [inline config/environments/development.rb]: convention

**Pull request merged**